packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -416,14 +416,38 @@ class Localizations extends StatefulWidget {
   /// true, in which case it returns null.
   static Locale? localeOf(BuildContext context, { bool nullOk = false }) {
     assert(context != null);
-    assert(nullOk != null);
     final _LocalizationsScope? scope = context.dependOnInheritedWidgetOfExactType<_LocalizationsScope>();
     if (nullOk && scope == null)
       return null;
-    assert(scope != null, 'a Localizations ancestor was not found');
+    assert(() {
+      if (scope == null) {
+        throw FlutterError(
+          'Localizations operation requested with a context that does not include Localizations.\n'
+          'The context used to retrieve the Localizations widget must be that of a widget that '
```

Comment:
This is slightly (very slightly) confusing, since the caller was trying to look up the current Locale. Might be helpful to weave Locale into the message. Likewise for the first line in this FlutterError.

Reply:
Updated wording here too.